bindata/network/ovn-kubernetes/ovnkube-node-smartnic-host.yaml

```diff
@@ -0,0 +1,315 @@
+---
```

Comment:
> Ugh. So much duplication between this file and `ovnkube-node.yaml`. Which also makes it hard to see what is actually different between the two...
> 
> Is there some way we could make the go code process `ovnkube-node.yaml` twice with different variables? 

I think yes, we want to process the `ovnkube-node.yaml` twice when `network.operator.openshift.io/ovn-offload-host` is labeled to any of the nodes in the cluster. perhaps we can list all nodes and detect the node labels, return a `smart-nic-enabled` flag in bootstrapOVN, which indicates that `renderOVNKubernetes` shall render the `ovnkube-node.yaml` twice, one in the `full` mode, the other in the `smart-nic-host` mode. wdyt?

Reply:
> I think yes, we want to process the `ovnkube-node.yaml` twice when `network.operator.openshift.io/ovn-offload-host` is labeled to any of the nodes in the cluster.

Hm... it seems weird to me to generate different outputs based on how nodes are labeled... I guess it's not exactly _forbidden_, but in general, we only vary our outputs based on config objects, not based on annotations on random other objects in the cluster... (And especially, this means that when CNO first starts up, it would only generate the "mode=full" configuration, because none of the masters would be labeled for offload, and then later when the workers got added to the cluster, it would have to generate the "mode=smart-nic-host" config as well...

OTOH, maybe the answer is just that there should be some flag in some config object that has to be enabled if you're going to use bf-2s...

Reply:
> > I think yes, we want to process the `ovnkube-node.yaml` twice when `network.operator.openshift.io/ovn-offload-host` is labeled to any of the nodes in the cluster.
> 
> Hm... it seems weird to me to generate different outputs based on how nodes are labeled... I guess it's not exactly _forbidden_, but in general, we only vary our outputs based on config objects, not based on annotations on random other objects in the cluster... (And especially, this means that when CNO first starts up, it would only generate the "mode=full" configuration, because none of the masters would be labeled for offload, and then later when the workers got added to the cluster, it would have to generate the "mode=smart-nic-host" config as well...
> 
> OTOH, maybe the answer is just that there should be some flag in some config object that has to be enabled if you're going to use bf-2s...


Agree. I think this is the first time we need to distinguish the nodes (config) in CNO because not all nodes in the cluster will be enabled with DPU. This means a single boolean flag in the CNO operator config will not be sufficient, we also need to support specifying the node group (e.g. MCP) which has DPU installed. Do you see we are adding this node group related config in CNO operator?